.github/workflows/e2e_test_for_prs.yml

```diff
@@ -0,0 +1,59 @@
+name: Run e2e for topic branch
+on:
+  pull_request:
+    paths-ignore:
+      - '**.md'
+      - 'cloudformation/*.yml'
+
+jobs:
+  run-e2e-for-topic-branch:
+    permissions:
+      id-token: write # This is required for requesting the JWT
+      contents: read  # This is required for actions/checkout
+    name: Run e2e for topic branch
+    runs-on: ubuntu-latest
+    steps:
+      - name: Git clone
+        uses: actions/checkout@v4
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@e3dd6a429d7300a6a4c196c26e071d42e0343502 #v4.0.2
+        with:
+          role-to-assume: ${{secrets.AWS_E2E_ROLE_ARN}}
+          aws-region: ${{vars.AWS_E2E_REGION}}
+      - name: Trigger AWS CodePipeline
+        run: |
+          git_hash=${{ github.event.pull_request.head.sha }}
+          echo $git_hash
+          codepipeline_execution_id=$(aws codepipeline start-pipeline-execution --name ${{vars.AWS_E2E_TOPIC_BRANCH_CODEPIPELINE_NAME}} --source-revisions actionName=Source,revisionType=COMMIT_ID,revisionValue=$git_hash --query 'pipelineExecutionId' --output text)
+          echo "Pipeline execution ID: $codepipeline_execution_id"
+          echo "codepipeline_execution_id=$codepipeline_execution_id" >> $GITHUB_ENV
+      - name: Poll Pipeline Status
+        id: poll-status
+        run: |
+          while true; do
+            sleep 30
```

Comment:
Why do we definitely wait for 30 seconds? Don't you think it's too long for a single iteration?

Reply:
e2e tests typically take 8-9 mins to complete. Checking each 30 seconds seemed appropriate.